### PR TITLE
firewall in use

### PR DESCRIPTION
**Pull request by mattkasun**
## Describe your changes
- correct detect firewall in use vice defaulting to iptables 
- detect if new firewall package installed (ie. iptables-nft) during checkin

## Provide testing steps
see NET-17

## Checklist before requesting a review
- [x] My changes affect only 10 files or less.
- [x] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [ ] If it is a bugfix, my code is <= 200 lines.
- [x] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [x] My unit tests pass locally.
- [ ] Netclient & Netmaker are awesome.

### CODE REVIEW

File: functions/mqpublish.go [hunk 1]
```diff
@@ -61,15 +61,11 @@ func Checkin(ctx context.Context, wg *sync.WaitGroup) {
...
 func checkin() {
-
+	// check/update host settings; publish if changed
 	if err := UpdateHostSettings(); err != nil {
-		logger.Log(0, "failed to update host settings -", err.Error())
+		logger.Log(0, "checkin", err.Error())
 		return
 	}
-
-	if err := PublishGlobalHostUpdate(models.HostMqAction(models.CheckIn)); err != nil {
```
Comment (abhishek9686) [inline functions/mqpublish.go:70]: why is this removed?
if checkin never occurs all the nodes will go into error state...

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

File: functions/mqpublish.go (hunk 1 above)
Comment (mattkasun) [reply functions/mqpublish.go:70]: UpdateHostSetttings calls PublishGlobalHostUpdate so leaving this would result in two identical messages being published.

Comment (expertqalityassurance) [reply functions/mqpublish.go:70]: Testing
as per the Abhishek comments we are facing first time warning issue and after some time it was showing error status also please check attachment

screenshot: 
1.https://prnt.sc/TRsSgZF4tEPL
2.https://prnt.sc/5YhDLi-6mZ4E
3.https://prnt.sc/OZJD52D4CDm6

Comment (abhishek9686) [reply functions/mqpublish.go:70]: > UpdateHostSetttings calls PublishGlobalHostUpdate so leaving this would result in two identical messages being published.
UpdateHostSetttings will only publish message when there is something that has changed, this won't accommodate `checkin` functionality

Comment (mattkasun) [reply functions/mqpublish.go:70]: removed PublishGlobalHostUpdate from UpdateHostSetting

File: config/config.go [hunk 2]
```diff
@@ -703,3 +694,32 @@ func IsHostInetGateway() bool {
 	}
 	return false
 }
+
+// setFirewall - determine and record firewall in use
+func SetFirewall() {
+	if ncutils.IsLinux() {
+		if ncutils.IsIPTablesPresent() {
+			netclient.FirewallInUse = models.FIREWALL_IPTABLES
```
Comment (afeiszli) [inline config/config.go:702]: What is the result of reporting FIREWALL_NONE if an ingress or egress is created? Will it be prevented, or just fail?

Comment (mattkasun) [reply config/config.go:702]: Create Ingress will return error. Create Egress will not  (ticket NET-20)

**Comment by expertqalityassurance** [comment]
Still I am facing same issue there are all nodes status is showing error
please check the below screenshot
 https://prnt.sc/g9tWSns0haK_

File: functions/mqpublish.go [hunk 3]
```diff
@@ -61,14 +61,11 @@ func Checkin(ctx context.Context, wg *sync.WaitGroup) {
...
-	if err := UpdateHostSettings(); err != nil {
-		logger.Log(0, "failed to update host settings -", err.Error())
-		return
-	}
-
-	if err := PublishGlobalHostUpdate(models.HostMqAction(models.CheckIn)); err != nil {
-		logger.Log(0, "failed to check-in", err.Error())
+	// check/update host settings; publish if changed
+	if UpdateHostSettings() {
+		if err := PublishGlobalHostUpdate(models.HostMqAction(models.CheckIn)); err != nil {
```
Comment (abhishek9686) [inline functions/mqpublish.go:66]: these are two different host update actions `checkin action` and `update action`, check-in should be sent irrespective of whether there is a change or not
can we just revert these changes to how it was previously?

**Review by abhishek9686 — CHANGES_REQUESTED** [review]

File: functions/mqpublish.go (hunk 3 above)
Comment (mattkasun) [reply functions/mqpublish.go]: reverted

**Review by theguy951357 — APPROVED** [review]
pr passes
critical list passes

**Review by abhishek9686 — APPROVED** [review]

**Review by afeiszli — APPROVED** [review]
